### PR TITLE
RG-2289 replace @nest with standardized nesting syntax

### PR DESCRIPTION
**Pull request by vkrol**

### CODE REVIEW

File: stylelint.config.js [hunk 1]
```diff
@@ -1,6 +1,9 @@
 module.exports = {
   extends: '@jetbrains/stylelint-config',
+  plugins: ['@csstools/stylelint-no-at-nest-rule'],
   rules: {
+    '@csstools/stylelint-no-at-nest-rule': true,
```
Comment (vkrol) [inline stylelint.config.js:5]: It's up for discussion :)

File: src/heading/heading.css [hunk 2]
```diff
@@ -9,30 +9,30 @@
     margin-top: 0;
   }
 
-  @nest h1&, h2& {
+  &:is(h1, h2) {
```
Comment (vkrol) [inline src/heading/heading.css:12]: Note that the native syntax does not allow us to write it like this:
```css
.heading {
  &h1 {}
}
```

File: stylelint.config.js (hunk 1 above)
Comment (andrey-skl) [reply stylelint.config.js:5]: Since it protects us from unwanted syntax usage, I don't see reason against that rule : )

**Review by andrey-skl — APPROVED** [review]

File: src/tags-input/tags-input.css [hunk 3]
```diff
@@ -61,7 +61,7 @@
 
   caret-color: var(--ring-main-color);
 
-  @nest [dir=rtl] & {
+  [dir="rtl"] & {
```
Comment (vkrol) [inline src/tags-input/tags-input.css:64]: Oops, it should be:
```suggestion
  [dir=rtl] & {
```
I will fix it.

**Pull request merged**